src/sentry/api/base.py

```diff
@@ -202,6 +204,30 @@ def paginate(self, request, on_results=None, paginator_cls=Paginator,
 
         return Response(results, headers=headers)
 
+    def bail_on_xorg(self, request, obj):
+        """
+        Will raise `ResourceDoesNotExist` when an authenticatee is attempting
+        to access a resource that is defined in another organization.
+        See: https://github.com/getsentry/sentry/issues/3807
+        """
+        if isinstance(obj, Organization):
+            org = obj
+        else:
+            if isinstance(obj, basestring):
```

Comment:
I'm not confident this check is safe. It's completely possible to use a primary key lookup with a string.

`Organization.objects.get(pk='1')` would do the right thing.

Can we just not be explicit on the `filters` through the function?

```python
bail_on_xorg(request, id=1)
bail_on_xorg(request, slug='foo')
bail_on_xorg(request, org)
```

That API seems chill, then leaves no ambiguity rather than trying to make an educated guess.